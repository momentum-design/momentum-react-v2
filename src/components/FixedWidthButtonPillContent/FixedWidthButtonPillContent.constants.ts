import { SIZES as BUTTON_PILL_SIZES } from '../ButtonPill/ButtonPill.constants';
import { SIZES as ICON_SIZES } from '../Icon/Icon.constants';

const CLASS_PREFIX = 'md-fixed-width-button-pill-content';

const DEFAULTS = {
  ICON: false,
  ICON_SCALE: ICON_SIZES[32],
  BUTTON_PILL_SIZE: BUTTON_PILL_SIZES[40],
};

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
  buttonContent: `${CLASS_PREFIX}-button-content`,
  hidden: `${CLASS_PREFIX}-hidden`,
};

export { CLASS_PREFIX, DEFAULTS, STYLE };
