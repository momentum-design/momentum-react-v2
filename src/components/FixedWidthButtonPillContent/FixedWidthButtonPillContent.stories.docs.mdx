The `<FixedWidthButtonPillContent />` component.

This component can be used if you have a `<ButtonPill />`, or a `<ButtonPillToggle />`, component
where the string content changes onPress or as a consequence of a state change. A button which
changes width in these circumstances may change the layout of a page and result in a suboptimal UX.

Tips:

We recommend adding `overflow: hidden;` to the button you use this with. Otherwise the user may be
able to trigger the onPress callback on the button from above the button.

Internally this component uses the Momentum React V2 `<Icon />` and `<Text />` components to mimic
the button content to achieve the fixed width. We recommend using the `<Text />` component for the
string child component to ensure truncation works as expected if you set a maximum width to the
button.
