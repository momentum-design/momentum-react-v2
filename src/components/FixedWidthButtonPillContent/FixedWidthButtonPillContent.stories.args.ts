import { commonStyles } from '../../storybook/helper.stories.argtypes';
import { BUTTON_PILL_CONSTANTS } from '../ButtonPill';
import { DEFAULTS } from './FixedWidthButtonPillContent.constants';
import { SIZES as ICON_SIZES } from '../Icon/Icon.constants';

const fixedWidthButtonPillContentArgTypes = {
  // NOTE: Below is an example. See [Storybook argTypes documentation]{@link https://storybook.js.org/docs/react/api/argtypes}.
  children: {
    description: 'Provides the child nodes for this element.', // NOTE: Description of this prop.
    control: { type: 'text' }, // NOTE: Control type for this prop.
    table: {
      type: {
        summary: 'ReactNode', // NOTE: Explicit type of this prop.
      },
      defaultValue: {
        summary: 'undefined', // NOTE: Default value for this prop.
      },
    },
  },
  buttonPillSize: {
    description: 'Modifies the size of the content to fit in any given `<ButtonPill />` component.',
    options: [...Object.values(BUTTON_PILL_CONSTANTS.SIZES)],
    control: { type: 'select' },
    table: {
      type: {
        summary: 'number',
      },
      defaultValue: {
        summary: BUTTON_PILL_CONSTANTS.DEFAULTS.SIZE,
      },
    },
  },
  icon: {
    description: 'Whether the content of this component will contain an `<Icon />` component.',
    options: [true, false],
    control: { type: 'boolean' },
    table: {
      type: {
        summary: 'boolean',
      },
      defaultValue: {
        summary: DEFAULTS.ICON,
      },
    },
  },
  iconScale: {
    description: 'The size of the `<Icon />` component if there is one.',
    options: [...Object.values(ICON_SIZES)],
    control: { type: 'select' },
    table: {
      type: {
        summary: 'number',
      },
      defaultValue: {
        summary: DEFAULTS.ICON_SCALE,
      },
    },
  },
  stringContentVariations: {
    description:
      'An array of strings that will be used in the `<ButtonPill />` component. The longest string will dictate the fixed width of the `<ButtonPill />`',
    // options: [true, false],
    control: { type: 'text' },
    table: {
      type: {
        summary: 'string[]',
      },
      defaultValue: {
        summary: ['Long, longer, longest'],
      },
    },
  },
};

export { fixedWidthButtonPillContentArgTypes };

export default {
  ...commonStyles,
  ...fixedWidthButtonPillContentArgTypes,
};
