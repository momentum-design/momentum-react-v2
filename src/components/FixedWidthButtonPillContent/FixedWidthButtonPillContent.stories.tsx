/* eslint-disable @typescript-eslint/no-empty-function */
import React, { useCallback, useState } from 'react';
import { Template } from '../../storybook/helper.stories.templates';
import { DocumentationPage } from '../../storybook/helper.stories.docs';
import StyleDocs from '../../storybook/docs.stories.style.mdx';

import FixedWidthButtonPillContent, { FixedWidthButtonPillContentProps } from './';
import argTypes from './FixedWidthButtonPillContent.stories.args';
import Documentation from './FixedWidthButtonPillContent.stories.docs.mdx';
import Icon from '../Icon';
import Text from '../Text';
import ButtonPill from '../ButtonPill';
import ButtonGroup from '../ButtonGroup';
import ButtonCircle from '../ButtonCircle';
import { getTextTypeFromButtonPillSize } from './FixedWidthButtonPillContent.utils';

export default {
  title: 'Momentum UI/FixedWidthButtonPillContent',
  component: FixedWidthButtonPillContent,
  parameters: {
    expanded: true,
    docs: {
      page: DocumentationPage(Documentation, StyleDocs),
    },
  },
};

const Common = Template<FixedWidthButtonPillContentProps>(() => {
  const englishStringContentVariations = ['Stop video', 'Start video'];
  const germanStringContentVariations = ['Stummschalten', 'Stummschaltung aufheben'];
  const englishChildren = [
    <>
      <Icon name="camera-on-colored" scale={20} />
      <Text>{englishStringContentVariations[0]}</Text>
    </>,
    <>
      <Icon
        name="camera-muted"
        scale={20}
        style={{ color: 'var(--mds-color-theme-text-error-normal)' }}
      />
      <Text>{englishStringContentVariations[1]}</Text>
    </>,
  ];
  const germanChildren = [
    <>
      <Icon name="audio-microphone-on-colored" scale={20} />
      <Text>{germanStringContentVariations[0]}</Text>
    </>,
    <>
      <Icon
        name="microphone-muted"
        scale={20}
        style={{ color: 'var(--mds-color-theme-text-error-normal)' }}
      />
      <Text>{germanStringContentVariations[1]}</Text>
    </>,
  ];
  const buttonPillStyle = { overflow: 'hidden', maxWidth: '12.375rem' };
  const [child, setChild] = useState(0);
  const handlePress = useCallback(() => {
    setChild((prev) => (prev + 1) % englishChildren.length);
  }, []);

  return (
    <div
      style={{
        display: 'grid',
        gridTemplateColumns: 'repeat(3, 15rem)',
        gridTemplateRows: 'repeat(3, auto)',
        gap: '0.5rem',
      }}
    >
      <ButtonPill
        style={{ width: '6rem', justifyContent: 'center' }}
        size={30}
        onPress={handlePress}
      >
        Press me
      </ButtonPill>
      <Text>English</Text>
      <Text>German</Text>
      <Text>With</Text>
      <ButtonGroup round compressed>
        <ButtonPill onPress={() => {}} ghost outline style={buttonPillStyle}>
          <FixedWidthButtonPillContent
            stringContentVariations={englishStringContentVariations}
            icon
            iconScale={20}
          >
            {englishChildren[child]}
          </FixedWidthButtonPillContent>
        </ButtonPill>
        <ButtonCircle outline ghost size={40}>
          <Icon data-test="arrow-icon" name="arrow-down" scale={12} weight="filled" />
        </ButtonCircle>
      </ButtonGroup>
      <ButtonGroup round compressed>
        <ButtonPill onPress={() => {}} ghost outline style={buttonPillStyle}>
          <FixedWidthButtonPillContent
            stringContentVariations={germanStringContentVariations}
            icon
            iconScale={12}
          >
            {germanChildren[child]}
          </FixedWidthButtonPillContent>
        </ButtonPill>
        <ButtonCircle outline ghost size={40}>
          <Icon data-test="arrow-icon" name="arrow-down" scale={12} weight="filled" />
        </ButtonCircle>
      </ButtonGroup>
      <Text>Without</Text>
      <ButtonGroup round compressed>
        <ButtonPill onPress={() => {}} ghost outline style={buttonPillStyle}>
          {englishChildren[child]}
        </ButtonPill>
        <ButtonCircle outline ghost size={40}>
          <Icon data-test="arrow-icon" name="arrow-down" scale={12} weight="filled" />
        </ButtonCircle>
      </ButtonGroup>
      <ButtonGroup round compressed>
        <ButtonPill onPress={() => {}} ghost outline style={buttonPillStyle}>
          {germanChildren[child]}
        </ButtonPill>
        <ButtonCircle outline ghost size={40}>
          <Icon data-test="arrow-icon" name="arrow-down" scale={12} weight="filled" />
        </ButtonCircle>
      </ButtonGroup>
    </div>
  );
}).bind({});

Common.argTypes = { ...argTypes };
delete Common.argTypes.children;
delete Common.argTypes.stringContentVariations;
delete Common.argTypes.icon;
delete Common.argTypes.iconScale;
delete Common.argTypes.buttonPillSize;

const Example = Template<FixedWidthButtonPillContentProps>((args) => {
  const { icon, iconScale, buttonPillSize } = args as FixedWidthButtonPillContentProps;
  const stringContentVariations = ['Long', 'Longer', 'Longest'];
  const [currentChildIndex, setCurrentChildIndex] = useState(0);
  const handlePress = useCallback(() => {
    setCurrentChildIndex((prev) => (prev + 1) % stringContentVariations.length);
  }, []);
  const buttonPillStyle = { overflow: 'hidden', maxWidth: '12.375rem' };
  const textType = getTextTypeFromButtonPillSize(buttonPillSize);

  return (
    <>
      <ButtonPill
        style={{ width: '6rem', justifyContent: 'center' }}
        size={30}
        onPress={handlePress}
      >
        Press me
      </ButtonPill>
      <div
        style={{
          display: 'grid',
          gridTemplateColumns: 'repeat(2, 15rem)',
          gridTemplateRows: 'repeat(2, auto)',
          gap: '0.5rem',
          marginTop: '0.5rem',
        }}
      >
        <Text>With</Text>
        <div>
          <ButtonPill
            onPress={handlePress}
            ghost
            outline
            size={buttonPillSize}
            style={buttonPillStyle}
          >
            <FixedWidthButtonPillContent
              stringContentVariations={stringContentVariations}
              icon={icon}
              iconScale={iconScale}
              buttonPillSize={buttonPillSize}
            >
              {icon && <Icon name="placeholder" scale={iconScale} />}
              <Text type={textType}>{stringContentVariations[currentChildIndex]}</Text>
            </FixedWidthButtonPillContent>
          </ButtonPill>
        </div>
        <Text>Without</Text>
        <div>
          <ButtonPill
            onPress={handlePress}
            ghost
            outline
            size={buttonPillSize}
            style={buttonPillStyle}
          >
            {icon && <Icon name="placeholder" scale={iconScale} />}
            <Text type={textType}>{stringContentVariations[currentChildIndex]}</Text>
          </ButtonPill>
        </div>
      </div>
    </>
  );
}).bind({});

Example.argTypes = { ...argTypes };
delete Example.argTypes.children;
delete Example.argTypes.stringContentVariations;

Example.args = {
  icon: false,
  iconScale: 20,
  buttonPillSize: 40,
};

export { Example, Common };
