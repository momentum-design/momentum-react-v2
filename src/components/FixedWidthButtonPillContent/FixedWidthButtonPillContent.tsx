import React, { FC } from 'react';
import classnames from 'classnames';

import { STYLE, DEFAULTS } from './FixedWidthButtonPillContent.constants';
import { Props } from './FixedWidthButtonPillContent.types';
import './FixedWidthButtonPillContent.style.scss';
import Icon from '../Icon';
import Text from '../Text';
import { getTextTypeFromButtonPillSize } from './FixedWidthButtonPillContent.utils';

/**
 * The FixedWidthButtonPillContent component.
 */
const FixedWidthButtonPillContent: FC<Props> = (props: Props) => {
  const {
    className,
    stringContentVariations,
    icon = DEFAULTS.ICON,
    iconScale = DEFAULTS.ICON_SCALE,
    buttonPillSize = DEFAULTS.BUTTON_PILL_SIZE,
    children,
  } = props;
  const textType = getTextTypeFromButtonPillSize(buttonPillSize);

  return (
    <div className={STYLE.wrapper}>
      {stringContentVariations.length &&
        stringContentVariations.map((string) => (
          <div
            key={string}
            className={classnames(STYLE.buttonContent)}
            data-size={buttonPillSize}
            data-hidden={true}
          >
            {icon && <Icon name="placeholder" scale={iconScale} />}
            <Text type={textType}>{string}</Text>
          </div>
        ))}
      <div
        className={classnames(className, STYLE.buttonContent)}
        data-size={buttonPillSize}
        data-hidden={false}
      >
        {children}
      </div>
    </div>
  );
};

export default FixedWidthButtonPillContent;
