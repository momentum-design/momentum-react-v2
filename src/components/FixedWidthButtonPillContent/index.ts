import { default as FixedWidthButtonPillContent } from './FixedWidthButtonPillContent';
import * as CONSTANTS from './FixedWidthButtonPillContent.constants';
import { Props } from './FixedWidthButtonPillContent.types';

export { CONSTANTS as FIXED_WIDTH_BUTTON_PILL_CONTENT_CONSTANTS };

export type FixedWidthButtonPillContentProps = Props;

export default FixedWidthButtonPillContent;
