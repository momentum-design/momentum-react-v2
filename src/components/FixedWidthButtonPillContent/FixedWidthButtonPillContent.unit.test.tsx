import React from 'react';
import { mount } from 'enzyme';

import FixedWidthButtonPillContent, {
  FIXED_WIDTH_BUTTON_PILL_CONTENT_CONSTANTS as CONSTANTS,
} from './';
import Icon from '../Icon';
import Text from '../Text';

describe('<FixedWidthButtonPillContent />', () => {
  describe('snapshot', () => {
    const snapshotChildren = (
      <>
        <Icon name="placeholder" scale={20} />
        <Text>Example</Text>
      </>
    );
    const stringContentVariationsExample = ['long', 'longer', 'longest'];
    const stringContentVariationsExample2 = ['test', 'testing'];

    it('should match snapshot', () => {
      expect.assertions(1);

      const container = mount(
        <FixedWidthButtonPillContent
          children={snapshotChildren}
          stringContentVariations={stringContentVariationsExample}
        />
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', () => {
      expect.assertions(1);

      const className = 'example-class';

      const container = mount(
        <FixedWidthButtonPillContent
          className={className}
          children={snapshotChildren}
          stringContentVariations={stringContentVariationsExample}
        />
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with stringContentVariations', () => {
      expect.assertions(1);

      const container = mount(
        <FixedWidthButtonPillContent
          children={snapshotChildren}
          stringContentVariations={stringContentVariationsExample2}
        />
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with buttonPillSize', () => {
      expect.assertions(1);

      const buttonPillSize = 24;

      const container = mount(
        <FixedWidthButtonPillContent
          buttonPillSize={buttonPillSize}
          children={snapshotChildren}
          stringContentVariations={stringContentVariationsExample}
        />
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with icon', () => {
      expect.assertions(1);

      const container = mount(
        <FixedWidthButtonPillContent
          icon
          children={snapshotChildren}
          stringContentVariations={stringContentVariationsExample}
        />
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with iconScale', () => {
      expect.assertions(1);

      const iconScale = 16;

      const container = mount(
        <FixedWidthButtonPillContent
          iconScale={iconScale}
          children={snapshotChildren}
          stringContentVariations={stringContentVariationsExample}
        />
      );

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with icon and iconScale', () => {
      expect.assertions(1);

      const iconScale = 16;

      const container = mount(
        <FixedWidthButtonPillContent
          icon
          iconScale={iconScale}
          children={snapshotChildren}
          stringContentVariations={stringContentVariationsExample}
        />
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    const children = (
      <>
        <Text>Example</Text>
      </>
    );
    const childrenWithIcon = (
      <>
        <Icon name="placeholder" scale={20} />
        <Text>Example</Text>
      </>
    );
    const stringContentVariationsExample = ['long', 'longer', 'longest'];

    it('should have its wrapper class', () => {
      expect.assertions(1);

      const element = mount(
        <FixedWidthButtonPillContent
          children={children}
          stringContentVariations={stringContentVariationsExample}
        />
      )
        .find(FixedWidthButtonPillContent)
        .getDOMNode();

      expect(element.classList.contains(CONSTANTS.STYLE.wrapper)).toBe(true);
    });

    it('should pass the buttonPillSize prop through correctly', () => {
      expect.assertions(4);

      const buttonPillSize = 32;

      const divElements = mount(
        <FixedWidthButtonPillContent
          children={children}
          stringContentVariations={stringContentVariationsExample}
          buttonPillSize={buttonPillSize}
        />
      ).find('div');

      for (let i = 1; i < 5; i++) {
        const divElement = divElements.at(i).getDOMNode();

        expect(divElement.getAttribute('data-size')).toEqual(`${buttonPillSize}`);
      }
    });

    it('should have provided class when className is provided', () => {
      expect.assertions(1);

      const className = 'class-name-example';

      const element = mount(
        <FixedWidthButtonPillContent
          className={className}
          children={children}
          stringContentVariations={stringContentVariationsExample}
        />
      )
        .find(FixedWidthButtonPillContent)
        .find('div')
        .at(4)
        .getDOMNode();

      expect(element.className.includes(className)).toBe(true);
    });

    it('should have a text component for each string content variation', () => {
      expect.assertions(3);

      const textElements = mount(
        <FixedWidthButtonPillContent
          children={children}
          stringContentVariations={stringContentVariationsExample}
        />
      )
        .find(FixedWidthButtonPillContent)
        .find(Text);

      const firstTextElementProps = textElements.at(0).props();
      const secondTextElementProps = textElements.at(1).props();
      const thirdTextElementProps = textElements.at(2).props();

      expect(firstTextElementProps).toStrictEqual({ type: 'subheader-primary', children: 'long' });
      expect(secondTextElementProps).toStrictEqual({
        type: 'subheader-primary',
        children: 'longer',
      });
      expect(thirdTextElementProps).toStrictEqual({
        type: 'subheader-primary',
        children: 'longest',
      });
    });

    it('should have text components with correct text type that corresponds to the button size', () => {
      expect.assertions(3);

      const textElements = mount(
        <FixedWidthButtonPillContent
          children={children}
          stringContentVariations={stringContentVariationsExample}
          buttonPillSize={24}
        />
      )
        .find(FixedWidthButtonPillContent)
        .find(Text);

      const firstTextElementProps = textElements.at(0).props();
      const secondTextElementProps = textElements.at(1).props();
      const thirdTextElementProps = textElements.at(2).props();

      expect(firstTextElementProps).toStrictEqual({ type: 'body-compact', children: 'long' });
      expect(secondTextElementProps).toStrictEqual({
        type: 'body-compact',
        children: 'longer',
      });
      expect(thirdTextElementProps).toStrictEqual({
        type: 'body-compact',
        children: 'longest',
      });
    });

    it('should have an icon component for each string content variation when icon = true ', () => {
      expect.assertions(3);

      const iconElements = mount(
        <FixedWidthButtonPillContent
          children={childrenWithIcon}
          stringContentVariations={stringContentVariationsExample}
          icon
        />
      )
        .find(FixedWidthButtonPillContent)
        .find(Icon);

      const firstIconElementProps = iconElements.at(0).props();
      const secondIconElementProps = iconElements.at(1).props();
      const thirdIconElementProps = iconElements.at(2).props();

      const iconProps = { name: 'placeholder', scale: 32 };

      expect(firstIconElementProps).toStrictEqual(iconProps);
      expect(secondIconElementProps).toStrictEqual(iconProps);
      expect(thirdIconElementProps).toStrictEqual(iconProps);
    });

    it('should have an icon component for each string content variation when icon = true with correct iconScale ', () => {
      expect.assertions(3);

      const iconScale = 16;

      const iconElements = mount(
        <FixedWidthButtonPillContent
          children={childrenWithIcon}
          stringContentVariations={stringContentVariationsExample}
          icon
          iconScale={iconScale}
        />
      )
        .find(FixedWidthButtonPillContent)
        .find(Icon);

      const firstIconElementProps = iconElements.at(0).props();
      const secondIconElementProps = iconElements.at(1).props();
      const thirdIconElementProps = iconElements.at(2).props();

      const iconProps = { name: 'placeholder', scale: iconScale };

      expect(firstIconElementProps).toStrictEqual(iconProps);
      expect(secondIconElementProps).toStrictEqual(iconProps);
      expect(thirdIconElementProps).toStrictEqual(iconProps);
    });

    it('should not have an icon component for each string content variation when icon = false ', () => {
      expect.assertions(1);

      const iconElements = mount(
        <FixedWidthButtonPillContent
          children={children}
          stringContentVariations={stringContentVariationsExample}
        />
      )
        .find(FixedWidthButtonPillContent)
        .find(Icon);

      expect(iconElements.length).toEqual(0);
    });

    it('should have children wrapped in a div with attribute data-hidden = false ', () => {
      expect.assertions(1);

      const element = mount(
        <FixedWidthButtonPillContent
          children={children}
          stringContentVariations={stringContentVariationsExample}
        />
      )
        .find('div')
        .at(4)
        .getDOMNode();

      expect(element.getAttribute('data-hidden')).toBe('false');
    });

    it('should have mimicked button content wrapped in a div with attribute data-hidden = true ', () => {
      expect.assertions(3);

      const buttonPillSize = 32;

      const divElements = mount(
        <FixedWidthButtonPillContent
          children={children}
          stringContentVariations={stringContentVariationsExample}
          buttonPillSize={buttonPillSize}
        />
      ).find('div');

      for (let i = 1; i < 4; i++) {
        const divElement = divElements.at(i).getDOMNode();

        expect(divElement.getAttribute('data-hidden')).toEqual('true');
      }
    });
  });
});
