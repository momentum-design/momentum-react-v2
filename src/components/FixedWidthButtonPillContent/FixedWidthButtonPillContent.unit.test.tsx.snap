// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FixedWidthButtonPillContent /> snapshot should match snapshot 1`] = `
<FixedWidthButtonPillContent
  stringContentVariations={
    Array [
      "long",
      "longer",
      "longest",
    ]
  }
>
  <div
    className="md-fixed-width-button-pill-content-wrapper"
  >
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={40}
      key="long"
    >
      <Text
        type="subheader-primary"
      >
        <h3
          className="md-text-wrapper"
          data-type="subheader-primary"
        >
          long
        </h3>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={40}
      key="longer"
    >
      <Text
        type="subheader-primary"
      >
        <h3
          className="md-text-wrapper"
          data-type="subheader-primary"
        >
          longer
        </h3>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={40}
      key="longest"
    >
      <Text
        type="subheader-primary"
      >
        <h3
          className="md-text-wrapper"
          data-type="subheader-primary"
        >
          longest
        </h3>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={false}
      data-size={40}
    >
      <Icon
        name="placeholder"
        scale={20}
      />
      <Text>
        <p
          className="md-text-wrapper"
          data-type="body-primary"
        >
          Example
        </p>
      </Text>
    </div>
  </div>
</FixedWidthButtonPillContent>
`;

exports[`<FixedWidthButtonPillContent /> snapshot should match snapshot with buttonPillSize 1`] = `
<FixedWidthButtonPillContent
  buttonPillSize={24}
  stringContentVariations={
    Array [
      "long",
      "longer",
      "longest",
    ]
  }
>
  <div
    className="md-fixed-width-button-pill-content-wrapper"
  >
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={24}
      key="long"
    >
      <Text
        type="body-compact"
      >
        <small
          className="md-text-wrapper"
          data-type="body-compact"
        >
          long
        </small>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={24}
      key="longer"
    >
      <Text
        type="body-compact"
      >
        <small
          className="md-text-wrapper"
          data-type="body-compact"
        >
          longer
        </small>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={24}
      key="longest"
    >
      <Text
        type="body-compact"
      >
        <small
          className="md-text-wrapper"
          data-type="body-compact"
        >
          longest
        </small>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={false}
      data-size={24}
    >
      <Icon
        name="placeholder"
        scale={20}
      />
      <Text>
        <p
          className="md-text-wrapper"
          data-type="body-primary"
        >
          Example
        </p>
      </Text>
    </div>
  </div>
</FixedWidthButtonPillContent>
`;

exports[`<FixedWidthButtonPillContent /> snapshot should match snapshot with className 1`] = `
<FixedWidthButtonPillContent
  className="example-class"
  stringContentVariations={
    Array [
      "long",
      "longer",
      "longest",
    ]
  }
>
  <div
    className="md-fixed-width-button-pill-content-wrapper"
  >
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={40}
      key="long"
    >
      <Text
        type="subheader-primary"
      >
        <h3
          className="md-text-wrapper"
          data-type="subheader-primary"
        >
          long
        </h3>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={40}
      key="longer"
    >
      <Text
        type="subheader-primary"
      >
        <h3
          className="md-text-wrapper"
          data-type="subheader-primary"
        >
          longer
        </h3>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={40}
      key="longest"
    >
      <Text
        type="subheader-primary"
      >
        <h3
          className="md-text-wrapper"
          data-type="subheader-primary"
        >
          longest
        </h3>
      </Text>
    </div>
    <div
      className="example-class md-fixed-width-button-pill-content-button-content"
      data-hidden={false}
      data-size={40}
    >
      <Icon
        name="placeholder"
        scale={20}
      />
      <Text>
        <p
          className="md-text-wrapper"
          data-type="body-primary"
        >
          Example
        </p>
      </Text>
    </div>
  </div>
</FixedWidthButtonPillContent>
`;

exports[`<FixedWidthButtonPillContent /> snapshot should match snapshot with icon 1`] = `
<FixedWidthButtonPillContent
  icon={true}
  stringContentVariations={
    Array [
      "long",
      "longer",
      "longest",
    ]
  }
>
  <div
    className="md-fixed-width-button-pill-content-wrapper"
  >
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={40}
      key="long"
    >
      <Icon
        name="placeholder"
        scale={32}
      />
      <Text
        type="subheader-primary"
      >
        <h3
          className="md-text-wrapper"
          data-type="subheader-primary"
        >
          long
        </h3>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={40}
      key="longer"
    >
      <Icon
        name="placeholder"
        scale={32}
      />
      <Text
        type="subheader-primary"
      >
        <h3
          className="md-text-wrapper"
          data-type="subheader-primary"
        >
          longer
        </h3>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={40}
      key="longest"
    >
      <Icon
        name="placeholder"
        scale={32}
      />
      <Text
        type="subheader-primary"
      >
        <h3
          className="md-text-wrapper"
          data-type="subheader-primary"
        >
          longest
        </h3>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={false}
      data-size={40}
    >
      <Icon
        name="placeholder"
        scale={20}
      />
      <Text>
        <p
          className="md-text-wrapper"
          data-type="body-primary"
        >
          Example
        </p>
      </Text>
    </div>
  </div>
</FixedWidthButtonPillContent>
`;

exports[`<FixedWidthButtonPillContent /> snapshot should match snapshot with icon and iconScale 1`] = `
<FixedWidthButtonPillContent
  icon={true}
  iconScale={16}
  stringContentVariations={
    Array [
      "long",
      "longer",
      "longest",
    ]
  }
>
  <div
    className="md-fixed-width-button-pill-content-wrapper"
  >
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={40}
      key="long"
    >
      <Icon
        name="placeholder"
        scale={16}
      />
      <Text
        type="subheader-primary"
      >
        <h3
          className="md-text-wrapper"
          data-type="subheader-primary"
        >
          long
        </h3>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={40}
      key="longer"
    >
      <Icon
        name="placeholder"
        scale={16}
      />
      <Text
        type="subheader-primary"
      >
        <h3
          className="md-text-wrapper"
          data-type="subheader-primary"
        >
          longer
        </h3>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={40}
      key="longest"
    >
      <Icon
        name="placeholder"
        scale={16}
      />
      <Text
        type="subheader-primary"
      >
        <h3
          className="md-text-wrapper"
          data-type="subheader-primary"
        >
          longest
        </h3>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={false}
      data-size={40}
    >
      <Icon
        name="placeholder"
        scale={20}
      />
      <Text>
        <p
          className="md-text-wrapper"
          data-type="body-primary"
        >
          Example
        </p>
      </Text>
    </div>
  </div>
</FixedWidthButtonPillContent>
`;

exports[`<FixedWidthButtonPillContent /> snapshot should match snapshot with iconScale 1`] = `
<FixedWidthButtonPillContent
  iconScale={16}
  stringContentVariations={
    Array [
      "long",
      "longer",
      "longest",
    ]
  }
>
  <div
    className="md-fixed-width-button-pill-content-wrapper"
  >
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={40}
      key="long"
    >
      <Text
        type="subheader-primary"
      >
        <h3
          className="md-text-wrapper"
          data-type="subheader-primary"
        >
          long
        </h3>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={40}
      key="longer"
    >
      <Text
        type="subheader-primary"
      >
        <h3
          className="md-text-wrapper"
          data-type="subheader-primary"
        >
          longer
        </h3>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={40}
      key="longest"
    >
      <Text
        type="subheader-primary"
      >
        <h3
          className="md-text-wrapper"
          data-type="subheader-primary"
        >
          longest
        </h3>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={false}
      data-size={40}
    >
      <Icon
        name="placeholder"
        scale={20}
      />
      <Text>
        <p
          className="md-text-wrapper"
          data-type="body-primary"
        >
          Example
        </p>
      </Text>
    </div>
  </div>
</FixedWidthButtonPillContent>
`;

exports[`<FixedWidthButtonPillContent /> snapshot should match snapshot with stringContentVariations 1`] = `
<FixedWidthButtonPillContent
  stringContentVariations={
    Array [
      "test",
      "testing",
    ]
  }
>
  <div
    className="md-fixed-width-button-pill-content-wrapper"
  >
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={40}
      key="test"
    >
      <Text
        type="subheader-primary"
      >
        <h3
          className="md-text-wrapper"
          data-type="subheader-primary"
        >
          test
        </h3>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={true}
      data-size={40}
      key="testing"
    >
      <Text
        type="subheader-primary"
      >
        <h3
          className="md-text-wrapper"
          data-type="subheader-primary"
        >
          testing
        </h3>
      </Text>
    </div>
    <div
      className="md-fixed-width-button-pill-content-button-content"
      data-hidden={false}
      data-size={40}
    >
      <Icon
        name="placeholder"
        scale={20}
      />
      <Text>
        <p
          className="md-text-wrapper"
          data-type="body-primary"
        >
          Example
        </p>
      </Text>
    </div>
  </div>
</FixedWidthButtonPillContent>
`;
